Apps/phjira/jira_connector.py

```diff
@@ -1411,6 +1406,60 @@ def _handle_remove_watcher(self, param):
 
         return action_result.set_status(phantom.APP_SUCCESS, "Successfully removed the user from the watchers list of the issue ID: {0}".format(issue_id))
 
+    def _handle_get_attachments(self,param):
+        self.save_progress(JIRA_USING_BASE_URL, base_url=self._base_url)
+        self.save_progress(phantom.APP_PROG_CONNECTING_TO_ELLIPSES, self._host)
+
+        # Create the jira object
+        if (phantom.is_fail(self._create_jira_object())):
+            return self.get_status()
+
+        action_result = self.add_action_result(phantom.ActionResult(dict(param)))
+
+        ticket_key = param['id']
+        get_all_attachments = param['retrieve_all']
+        extension_filter = param.get('extension_filter')
+        container_id = param['container_id']
+
+        try:
+            jira_issue = self._jira.issue(ticket_key, expand="attachment")
+        except Exception as e:
+            return action_result.set_status(phantom.APP_ERROR, "Please enter a valid Jira Ticket ID. Please check the provided parameters.")
+
+        if len(jira_issue.fields.attachment) > 0:
+            temp_vault_path = Vault.get_vault_tmp_dir().rstrip('/')
+            if get_all_attachments:
+                for attachment in jira_issue.fields.attachment:
+                    jira_filename = "".join(str(attachment.filename).split())
+                    full_path = '{}/{}'.format(temp_vault_path,jira_filename)
+                    with open(full_path, 'wb') as f:
+                        attachment_content = attachment.get()
+                        f.write(attachment_content)
+                        action_result.add_data(Vault.add_attachment(file_location=full_path,container_id=container_id))
+
+            elif extension_filter:
+                extension_list = extension_filter.split(',')
+                extension_list = [".{}".format(extension.lstrip('.')) for extension in extension_list]
+
+                for attachment in jira_issue.fields.attachment:
+                    jira_filename = "".join(str(attachment.filename).split())
+                    full_path = '{}/{}'.format(temp_vault_path, jira_filename)
+                    file_extension = ".{}".format(jira_filename.rsplit('.')[-1])
+
+                    if file_extension in extension_list:
+                        with open(full_path, 'wb') as f:
```

Comment:
Code duplication.  Only four lines, so not a lot but possibly could write a helper function for writing attachments.

Reply:
Created helper function _write_int_file().